### PR TITLE
Adding Methanol Synthesis Flowsheet Example Notebook

### PR DESCRIPTION
**Pull request by bpaul4**
## Proposed changes:
- Adding methanol synthesis example notebook to EXAMPLES-PSE repository

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by lbianchi-lbl** [comment]
@bpaul4 If you'd like to test this PR with IDAES/idaes-pse#711 before that one is merged into `idaes-pse:main`, you should be able to just set the proper value here: https://github.com/IDAES/examples-pse/blob/79dfed7ade2a352bf9d847bbd155ed40bf6d4559/.github/workflows/main.yml#L17, i.e.:

```yaml
env:
  IDAES_CONDA_ENV_NAME_DEV: examples-pse-dev
  IDAES_PSE_REQUIREMENT: idaes-pse[prerelease] @ git+https://github.com/IDAES/idaes-pse@refs/pull/711/head
``

File: src/Examples/Flowsheets/methanol_synthesis.ipynb [hunk 1]
```diff
@@ -0,0 +1,1749 @@
...
+    "print()\n",
+    "print('Solving optimization problem...')\n",
+    "opt_res = solver.solve(m, tee=True)\n",
+    "assert opt_res.solver.termination_condition == TerminationCondition.optimal\n",
+    "print('Optimal solution process results:')\n",
+    "report(m)"
+   ]
+  },
+  {
+   "cell_type": "code",
```
Comment (andrewlee94) [inline src/Examples/Flowsheets/methanol_synthesis.ipynb:1705]: It is hard to tell via the GitHub interface, but this appears to be missing the tags to indicate that this is a testing cell and should not be included in the public facing version,

**Review by andrewlee94 — CHANGES_REQUESTED** [review]
As a general comment, I think there needs to be more guidance to the user if this is to be a proper example. Each step should explain what is happening, why and what the expected outcome is.

**Comment by ksbeattie** [comment]
@andrewlee94, @MAZamarripa this is ready for re-review.

**Review by andrewlee94 — APPROVED** [review]
I took a quick look and only had a couple of minor observations:

1. A summary upfront to explain the purpose of the example (in a sentence or two) would probably be valuable. This would help inform users of whether this example is of interest to them.

2. Do we really need 500 iterations for IPOPT? Unless there is a good reason, I think we should be using a more reasonable value for this. Otherwise, readers will start to think 500 iterations is expected/acceptable, when really it should solve in less than 100 in most cases.

**Comment by bpaul4** [comment]
@MAZamarripa this is ready for your review when you have a moment. Thank you!

**Review by MAZamarripa — APPROVED** [review]
lgtm

**Pull request merged**